src/materials/MeshRefractionMaterial.tsx

```diff
@@ -16,41 +16,40 @@ export const MeshRefractionMaterial = shaderMaterial(
     bvh: new MeshBVHUniformStruct(),
     color: new THREE.Color('white'),
     resolution: new THREE.Vector2(),
+    viewMatrixInverse: new THREE.Matrix4(),
+    projectionMatrixInverse: new THREE.Matrix4(),
   },
   /*glsl*/ `
-  #ifndef USE_COLOR
-    uniform vec3 color;
-  #endif
+  uniform mat4 viewMatrixInverse;
+
   varying vec3 vWorldPosition;  
   varying vec3 vNormal;
-  varying mat4 projectionMatrixInv;
-  varying mat4 viewMatrixInv;
-  varying vec3 viewDirection;
-  varying mat4 vInstanceMatrix;
-  varying vec3 vColor;
-  
+  varying mat4 vModelMatrixInverse;
+
+  #ifdef USE_INSTANCING_COLOR
+    varying vec3 vInstanceColor;
+  #endif
+
   void main() {        
     vec4 transformedNormal = vec4(normal, 0.0);
     vec4 transformedPosition = vec4(position, 1.0);
     #ifdef USE_INSTANCING
-      vInstanceMatrix = instanceMatrix;
       transformedNormal = instanceMatrix * transformedNormal;
       transformedPosition = instanceMatrix * transformedPosition;
+    #endif
+
+    #ifdef USE_INSTANCING
+      vModelMatrixInverse = inverse(modelMatrix * vInstanceMatrix);
```

Comment:
for some additional savings you could pass in `inverse(modelMatrix)` as a uniform too 😄